Check commutativity for computed dispatch table and add a test to check entries.

ailzhang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44088 Check commutativity for computed dispatch table and add a test to check entries.**

Differential Revision: [D23492793](https://our.internmc.facebook.com/intern/diff/D23492793)

Commit:
Check commutativity for computed dispatch table and add a test to check entries.

[ghstack-poisoned]

ezyang commented on test/test_dispatch.py:
```diff
@@ -181,15 +188,16 @@ def go(ctor_order):
         # error in test construction: it means you could have
         # factored the test into two smaller ones.
-        return results[frozenset(range(len(ops)))][0]
+        return results[frozenset(range(len(ops)))][:-1]
```
It's not obvious to me that you get a 2-size tuple here. Also, consider returning a named tuple, so that you can avoid ugly `[0]` indexing.

ezyang commented on test/test_dispatch.py:
```diff
@@ -277,27 +285,61 @@ def test_impl_only(self):
+catchall: default_def_name_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+''')
+        # computed dispatch table is too big to use assertExpectedInline,
+        # so we only assert on a few entries we're interested in.
```
My preference is for us to instead specify a list of keys that we are interested in seeing in the computed dispatch table, and then use an expect test on it. That will make it easier for us to do wide-sweeping refactors to the dump format in the future.

ezyang commented:
Thanks, looking good!

Commit:
Update on "Check commutativity for computed dispatch table and add a test to check entries."


Differential Revision: [D23492793](https://our.internmc.facebook.com/intern/diff/D23492793)

[ghstack-poisoned]

Commit:
Update on "Check commutativity for computed dispatch table and add a test to check entries."


Differential Revision: [D23492793](https://our.internmc.facebook.com/intern/diff/D23492793)

[ghstack-poisoned]

ezyang commented on test/test_dispatch.py:
```diff
@@ -267,37 +279,87 @@ def test_impl_only(self):
+                for t in table_entries:
+                    if t.startswith(k):
+                        extracted += (t + '\n')
+            return extracted
```
You're probably going to want to use this in other entries later.

ailzhang replied:
Yup! will move it out when it's used for the second time :D
